native/membrane_videocompositor_opengl_rust/src/framebuffers.rs

```diff
@@ -0,0 +1,171 @@
+//! Structures for managing various OpenGL render targets
+use glad_gles2::gl;
+
+/// An abstraction for an individual [framebuffer object](<https://www.khronos.org/opengl/wiki/Framebuffer_Object>) with an attached [renderbuffer](<https://www.khronos.org/opengl/wiki/Renderbuffer_Object>).
+struct FramebufferObject {
+  width: usize,
+  height: usize,
+  id: gl::GLuint,
+  renderbuffer_id: gl::GLuint,
+  _internal_format: gl::GLenum, // FIXME these should be custom enums instead of GLenum random ints
+  format: gl::GLenum,
+  type_: gl::GLenum,
+}
+
+impl FramebufferObject {
+  /// Create a new instance.
+  ///  * `width` and `height` should be given in pixels.
+  ///  * `internal_format` is the format used by the renderbuffer internally (e.g `gl::RGB8` or `gl::R8`)
+  ///  * `format` and `type_` represent the type and format we expect to get out when reading from the framebuffer (e.g. `gl::RGB` for `format` and `gl::UNSIGNED_BYTE` for `type_`)
+  fn new(
+    width: usize,
+    height: usize,
+    internal_format: gl::GLenum,
+    format: gl::GLenum,
+    type_: gl::GLenum,
```

Comment:
```suggestion
    object_type: gl::GLenum,
```

or something similar; should end up more readable